### PR TITLE
prow: add support for clearing milestones

### PR DESCRIPTION
**Pull request by marun**
Posting this for @dankohn in support of https://github.com/kubernetes/test-infra/issues/5344

/area prow
/assign bentheelder
/assign cjwagner

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -1426,3 +1426,22 @@ func (c *Client) IsMergeable(org, repo string, number int, sha string) (bool, er
...
+
+// ClearMilestone clears the milestone from the specified issue
+func (c *Client) ClearMilestone(org, repo string, num int) error {
+	c.log("ClearMilestone", org, repo, num)
+	if c.fake {
+		return nil
+	}
+
+	issue := &struct {
+		Milestone interface{} `json:"milestone"`
```
Comment (stevekuznetsov) [inline prow/github/client.go]: Is this `interface{}` just because you want the serialized value to be `null` and don't need a type?

Comment (marun) [reply prow/github/client.go]: Correct.  afaik this is the only way to clear the milestone.

File: prow/github/client.go [hunk 2]
```diff
@@ -1426,3 +1426,22 @@ func (c *Client) IsMergeable(org, repo string, number int, sha string) (bool, er
 	}
 	return false, fmt.Errorf("reached maximum number of retries (%d) checking mergeability", maxTries)
 }
+
+// ClearMilestone clears the milestone from the specified issue
+func (c *Client) ClearMilestone(org, repo string, num int) error {
+	c.log("ClearMilestone", org, repo, num)
+	if c.fake {
```
Comment (cjwagner) [inline prow/github/client.go]: I don't think you need this check. If `c.fake` is true `c.request()` will react appropriately.

File: prow/github/client.go (hunk 1 above)
Comment (cjwagner) [reply prow/github/client.go]: Please add a short comment explaining this.

Comment (marun) [reply prow/github/client.go]: Done.

File: prow/github/client.go (hunk 2 above)
Comment (marun) [reply prow/github/client.go]: Done.

**Pull request merged**